manifests/params.pp

```diff
@@ -308,7 +308,7 @@
       # it reports its $osreleasemajor as 2, not 6.
       # thats why we're matching for '2' in both parts
       # Amazon Linux is like RHEL6 but reports its osreleasemajor as 2017 or 2018.
-      $osreleasemajor = regsubst($::operatingsystemrelease, '^(\d+)\..*$', '\1') # workaround for the possibly missing operatingsystemmajrelease
+      $osreleasemajor = regsubst($facts['os']['release']['full'], '^(\d+)\..*$', '\1') # workaround for the possibly missing operatingsystemmajrelease
```

Comment:
I'm not sure about this, https://github.com/theforeman/puppet-puppet/pull/694 was intended to solve it in a cleaner way.